experimental/packages/otlp-grpc-exporter-base/test/grpc-exporter-transport.test.ts

```diff
@@ -251,11 +276,62 @@ describe('GrpcExporterTransport', function () {
 
         // clear context
         serverTestContext.requests = [];
+        serverTestContext.metadata = [];
         serverTestContext.serverResponseProvider = () => {
           return { error: null, buffer: Buffer.from([]) };
         };
       });
 
+      const getUserAgent = () => {
+        return serverTestContext.metadata[0].get('user-agent')[0] as string;
+      };
```

Comment:
nit: I find it sometimes difficult to trace back where things are coming from in tests, so I like to inject it like so instead of relying on the current state.

```suggestion
      function getUserAgent (serverTestContext: ServerTestContext) {
        return serverTestContext.metadata[0].get('user-agent')[0] as string;
      };
```